packages/main/cypress/specs/Dialog.cy.ts

```diff
@@ -0,0 +1,413 @@
+import { html } from "lit";
+import "../../src/Dialog.js";
+import "../../src/Button.js";
+import "../../src/Input.js";
+
+describe("Dialog", () => {
+	describe("Dialog general interaction", () => {
+		beforeEach(() => {
+			cy.mount(html`
+				<ui5-button id="btnOpenDialog">Open Stretched Dialog</ui5-button>
+				<ui5-dialog id="dialog" header-text="Dialog">
+					<span>Some content</span>
+					<div slot="footer" class="dialogFooter">
+						<ui5-button id="btnCloseDialog">Close</ui5-button>
+					</div>
+				</ui5-dialog>
+			`);
+		});
+
+		it("tests dialog toggling", () => {
+			cy.get("#btnOpenDialog").then($btnOpenDialog => {
+				$btnOpenDialog.get(0).addEventListener("click", () => {
+					cy.get("[ui5-dialog]").invoke("prop", "open", true);
+				});
+			});
+
+			cy.get("#btnOpenDialog").realClick();
+			cy.get("[ui5-dialog]").should("be.visible");
+
+			cy.get("#btnCloseDialog").then($btnCloseDialog => {
+				$btnCloseDialog.get(0).addEventListener("click", () => {
+					cy.get("[ui5-dialog]").invoke("prop", "open", false);
+				});
+			});
+
+			cy.get("#btnCloseDialog").realClick();
+			cy.get("[ui5-dialog]").should("not.be.visible");
+		});
+
+		it("tests dialog toggling with 'open' attribute", () => {
+			cy.get("#btnOpenDialog").then($btnOpenDialog => {
+				$btnOpenDialog.get(0).addEventListener("click", () => {
+					cy.get("[ui5-dialog]").invoke("prop", "open", "''");
+				});
+			});
+
+			cy.get("#btnOpenDialog").click();
+			cy.get("[ui5-dialog]").should("be.visible");
+
+			cy.get("#btnCloseDialog").then($btnCloseDialog => {
+				$btnCloseDialog.get(0).addEventListener("click", () => {
+					cy.get("[ui5-dialog]").invoke("removeAttr", "open");
+				});
+			});
+
+			cy.get("#btnCloseDialog").realClick();
+			cy.get("[ui5-dialog]").should("not.be.visible");
+		});
+
+		it("dialog repositions after screen resize", () => {
+			// Setup
+			cy.get("#btnOpenDialog").then($btnOpenDialog => {
+				$btnOpenDialog.get(0).addEventListener("click", () => {
+					cy.get("[ui5-dialog]").invoke("prop", "open", true);
+				});
+			});
+
+			cy.get("#btnOpenDialog").realClick();
+
+			cy.get("[ui5-dialog]").then($dialog => {
+				const topBeforeScreenResize = parseInt($dialog[0].style.top);
+				const leftBeforeScreenResize = parseInt($dialog[0].style.left);
+
+				// Act
+				cy.viewport(600, 600).then(() => {
+					// Assert
+					cy.get("[ui5-dialog]").then($dialogRepositioned => {
+						setTimeout(() => {
+							const topAfterScreenResize = parseInt($dialogRepositioned[0].style.top);
+							const leftAfterScreenResize = parseInt($dialogRepositioned[0].style.left);
+
+							expect(topBeforeScreenResize).to.not.equal(topAfterScreenResize);
+							expect(leftBeforeScreenResize).to.not.equal(leftAfterScreenResize);
+						}, 1);
+					});
+				});
+
+				// Clean-up
+				cy.realPress("Escape");
```

Comment:
You don't need to clean up since a new dialog is used for every test and previous one is removed from the DOM.